healthcheck/healthcheck.go

```diff
@@ -0,0 +1,102 @@
+package healthcheck
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cloudwan/gohan/db"
+	"github.com/cloudwan/gohan/db/transaction"
+	"github.com/cloudwan/gohan/log"
+	"github.com/cloudwan/gohan/sync"
+	"github.com/cloudwan/gohan/util"
+)
+
+const (
+	healthCheckKey        = "healthcheck"
+	healthCheckEnabledKey = healthCheckKey + "/enabled"
+	healthCheckEtcdKey    = healthCheckKey + "/etcd_key"
+	healthCheckAddressKey = healthCheckKey + "/address"
+	healthCheckTimeoutKey = healthCheckKey + "/timeout"
+	defaultEtcdKey        = "/gohan"
+	defaultTimeoutInMs    = 5000
+)
+
+type HealthCheck struct {
+	DataStore db.DB
+	Etcd      sync.Sync
+	etcdKey   string
+	address   string
+	timeout   time.Duration
+	logger    log.Logger
+}
+
+func NewHealthCheck(db db.DB, etcd sync.Sync, serverAddress string, config *util.Config) *HealthCheck {
+	if enabled := config.GetBool(healthCheckEnabledKey, false); !enabled {
```

Comment:
this `enabled` seems redundant

Reply:
` if config.GetBool() {` should be enough.